common/build/eslint-config-fluid/scripts/print-configs.cjs

```diff
@@ -73,27 +73,47 @@ async function printConfig(filePath, configPath) {
 
 (async () => {
 	const args = process.argv.slice(2);
+	
+	if (args.length !== 1) {
+		console.error("Usage: node print-configs.cjs <output-directory>");
+		process.exit(1);
+	}
+	
 	const outputPath = args[0];
 	const writePromises = [];
+	const expectedFiles = new Set();
 
 	for (const { name, configPath, sourceFilePath } of configsToPrint) {
 		const outputFilePath = path.join(outputPath, `${name}.json`);
+		expectedFiles.add(`${name}.json`);
+		
 		let originalContent = "";
 		try {
 			originalContent = await fs.readFile(outputFilePath, "utf8");
 		} catch (err) {
-			console.error(`Error reading file ${outputFilePath}:`, err);
-			// It's OK to continue because we might be outputting a new file.
-			continue;
+			// File doesn't exist yet, which is OK - we'll create it
 		}
 
-		const newContent = await printConfig(sourceFilePath, configPath);
+		const newContent = await generateConfig(sourceFilePath, configPath);
 
 		// Only write the file if the content has changed
 		if (newContent !== originalContent) {
 			writePromises.push(fs.writeFile(outputFilePath, newContent));
 		}
 	}
 
+	// Remove any files in the output directory that aren't in the expected list
+	try {
+		const existingFiles = await fs.readdir(outputPath);
+		for (const file of existingFiles) {
+			if (file.endsWith(".json") && !expectedFiles.has(file)) {
+				console.log(`Removing unexpected file: ${file}`);
+				await fs.unlink(path.join(outputPath, file));
+			}
+		}
+	} catch (err) {
+		// Output directory might not exist yet, which is OK
+	}
```

Comment:
At this point the `outputPath` should always exist, no? The code above should have written files inside it. If there's a real scenario where we don't expect it, I'd say let's do an exists check first to avoid empty catch blocks.

Reply:
It's still not a safe operation to assume the path exists or has files. Are you suggesting we throw if readdir or unlink throw? I think we should only throw if writing an expected file fails.